library/cipher.c

```diff
@@ -218,6 +283,78 @@ int mbedtls_cipher_setkey( mbedtls_cipher_context_t *ctx,
     if( ctx->cipher_info == NULL )
         return( MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
 
+    if( operation != MBEDTLS_DECRYPT &&
```

Comment:
These checks have moved to parameter validation in `development`, so should be removed here I think.